src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs

```diff
@@ -208,6 +210,53 @@ public void Pop(bool success)
             SetConstructorArgumentState();
         }
 
+        /// <summary>
+        /// Configures the current stack frame for a polymorphic converter.
+        /// </summary>
+        public JsonConverter InitializePolymorphicReEntry(JsonTypeInfo derivedJsonTypeInfo)
+        {
+            Debug.Assert(!IsContinuation);
+            Debug.Assert(Current.PolymorphicJsonTypeInfo == null);
+            Debug.Assert(Current.PolymorphicSerializationState == PolymorphicSerializationState.None);
+
+            Current.PolymorphicJsonTypeInfo = Current.JsonTypeInfo;
+            Current.JsonTypeInfo = derivedJsonTypeInfo.PropertyInfoForTypeInfo.JsonTypeInfo;
+            Current.JsonPropertyInfo = Current.JsonTypeInfo.PropertyInfoForTypeInfo;
+            Current.NumberHandling ??= Current.JsonPropertyInfo.NumberHandling;
```

Comment:
doesn't this mean the base type will take config will take precedence over derived type? should this be:

```csharp
Current.NumberHandling = Current.JsonPropertyInfo.NumberHandling ?? Current.NumberHandling;
```

Reply:
or just simply `Current.NumberHandling = Current.JsonPropertyInfo.EffectiveNumberHandling`

Reply:
Good point, but that probably needs to consult the parent number handling as happens in the `Push` operation:
```C#
Current.NumberHandling = Parent.NumberHandling ?? Current.JsonPropertyInfo.EffectiveNumberHandling;
```
Strange, I thought I had changed this.

Reply:
I think `Current.JsonPropertyInfo.NumberHandling ?? parent.EffectiveNumberHandling` - note that EffectiveNumberHandling already takes options, declaring type etc into consideration